### PR TITLE
Various enhancements and fixes

### PR DESCRIPTION
**Pull request by SwoopX**
- Extend sensor fingerprints
- Improve sensor selection
- Initialize bool values for Thermostats with "false"
- Correct typo in modelID for edp re:dy smart plug (#3926)
- Enhance support for Stelpro baseboard thermostat STZB402+ (#4049)
- Add complete EcoDim and Namron ModelIDs to button maps (#4076)
- Add complete iCasa ModelID to button_maps.json (ICZB-RM11S) (#4125)
- Enhance debug output on valid button event
- Correct model ID in button maps for LZL4BWHL01 Remote (#4151)
- Show info on button presses in debug log for Philips switches
- Increase time resolution and add date for etag to decrease hash collisions (#4052)
- Streamline way of setting etags

### CODE REVIEW

File: de_web_plugin.cpp [hunk 1]
```diff
@@ -1816,9 +1816,9 @@ void DeRestPluginPrivate::updateEtag(QString &etag)
 {
     QTime time = QTime::currentTime();
 #if QT_VERSION < 0x050000
-    etag = QString(QCryptographicHash::hash(time.toString().toAscii(), QCryptographicHash::Md5).toHex());
+    etag = QString(QCryptographicHash::hash(time.toString("hh:mm:ss.zzz").toAscii(), QCryptographicHash::Md5).toHex());
```
Comment (hbkoenig) [inline de_web_plugin.cpp:1819]: why do you use *only* time for etag hash, why not ad date too ?
and why not add measurement values and/or other stuff too ?
the md5 hash is much longer than the raw time string (12 vs. 32 chars), I don't get it what here a md5 hash can be good for ?!?

File: diagnostics.cpp [hunk 2]
```diff
@@ -7,11 +7,11 @@
  */
 void DeRestPluginPrivate::handleDiagnosticsClusterIndication(const deCONZ::ApsDataIndication &ind, deCONZ::ZclFrame &zclFrame)
 {
-    Sensor *sensor = getSensorNodeForAddressAndEndpoint(ind.srcAddress(), ind.srcEndpoint());
+    Sensor *sensor = getSensorNodeForAddressAndEndpoint(ind.srcAddress(), ind.srcEndpoint(), QLatin1String("ZHAThermostat"));
 
     if (!sensor)
     {
-        DBG_Printf(DBG_INFO, "No sensor found for 0x%016llX, endpoint: 0x%08X\n", ind.srcAddress().ext(), ind.srcEndpoint());
+        DBG_Printf(DBG_INFO, "No thermostat sensor found for 0x%016llX, endpoint: 0x%08X\n", ind.srcAddress().ext(), ind.srcEndpoint());
```
Comment (manup) [inline diagnostics.cpp:14]: `handleDiagnosticsClusterIndication()` isn't this to be supposed to be general, why thermostat?

`endpoint: 0x%08X` shouldn't this be `endpoint: 0x%02X`?

File: air_quality.cpp [hunk 3]
```diff
@@ -140,8 +140,7 @@ void DeRestPluginPrivate::handleAirQualityClusterIndication(const deCONZ::ApsDat
 
         if (configUpdated || stateUpdated)
         {
-            updateEtag(sensor->etag);
-            updateEtag(gwConfigEtag);
```
Comment (manup) [inline air_quality.cpp:144]: Note the `gwConfigEtag` needs to be updated as well since querying the full config state is affected by this.

**Review by manup — COMMENTED** [review]
Added a few comments.

File: air_quality.cpp (hunk 3 above)
Comment (SwoopX) [reply air_quality.cpp:144]: Using `updateSensorEtag(&*sensor);` instead does take care of that, doesn't it?

File: diagnostics.cpp (hunk 2 above)
Comment (SwoopX) [reply diagnostics.cpp]: For the endpoint: Right, corrected that in other files as well

For the generic handling: Happy to remove the "thermostat" from the message. Regarding the generic usage, currently only 2 thermostats require this cluster. However, we allow further processing only if a sensor is available. And to pick the right one, I currently have to pin the sensor type some lines above.

**Pull request merged**